### DISCUSSION TITLE
Testes

### DISCUSSION THREAD

**Discussion by nitoba** [diego3g/rsxp-2023#14] [General]
Tratando-se do backend, devido a ser um MVP com duração máxima de até duas semanas, serão escritos testes para a aplicação ?
Sendo sim, a arquitetura do projeto deveria ser o mínimo testável possível. Devido a utilização do NestJs a aplicação já está mais preparada prós testes de maneira mais tranquila

**Comment by LuizHAP** [comment]
Acho super maneiro! No backend é mais fácil, mas nunca mexi com testes no mobile

**Reply by brunoesposito** [reply]
No app alem do teste isso pode ser escalado para lista de teste interno e externos dentro das stores se o tempo for curto pra realmente desenvolver lá!

**Reply by nitoba** [reply]
Você fala sobre fazer QA ao invés de fazer testes de unidade por exemplo ?

**Reply by LuizHAP** [reply]
Acho maneiro! Eu participaria de QA fácil!

**Reply by melquisedeque-magalhaes** [reply]
TMB acho uma boa ideia 💡, ter um time de QA, tanto no back quanto no front

**Comment by henriqgoncalvs** [comment]
Pra testes no app acho que o ideal seria focar em testes de integração para as telas. A gente poderia usar o [MSW](https://mswjs.io/) e o [RNTL](https://callstack.github.io/react-native-testing-library/) pra isso como eu comentei aqui https://github.com/diego3g/rsxp-2023/discussions/15#discussioncomment-5467770

**Comment by diego3g** [comment]
Concordo em seguir com testes mais alto nível nesse primeiro momento, focando em testes E2E e testes da comunidade mesmo. Acho que no React Native existe o Detox que podemos usar no back-end podemos seguir com testes de rotas mesmo via supertest.

**Comment by diego3g** [comment]
Vou fechar essa discussão porque estamos com bastante discussões abertas com tópicos que super valem a contribuição de vocês por lá também e acho que não vale estender muito a discussão por aqui. Sintam-se livres para reabrir a discussão se fizer sentido 🤗
